frontend/src/components/parties/PartyProfile.js

```diff
@@ -36,112 +37,108 @@ const propTypes = {
   fetchMineBasicInfoList: PropTypes.func.isRequired,
   parties: PropTypes.arrayOf(CustomPropTypes.party).isRequired,
   partyRelationships: PropTypes.arrayOf(CustomPropTypes.partyRelationship),
-  partyRelationshipTypes: PropTypes.arrayOf(CustomPropTypes.dropdownListItem),
-  mineBasicInfoList: PropTypes.arrayOf(CustomPropTypes.mine),
-  match: PropTypes.object,
+  partyRelationshipTypeHash: PropTypes.objectOf(PropTypes.strings),
+  mineBasicInfoListHash: PropTypes.objectOf(PropTypes.strings),
+  match: CustomPropTypes.match.isRequired,
 };
 
 const defaultProps = {
   partyRelationships: [],
-  partyRelationshipTypes: [],
-  mineBasicInfoList: [],
+  partyRelationshipTypeHash: {},
+  mineBasicInfoListHash: {},
 };
 
 export class PartyProfile extends Component {
+  state = { isLoaded: false };
+
   componentDidMount() {
     const { id } = this.props.match.params;
     this.props.fetchPartyById(id);
     this.props.fetchPartyRelationships({ party_guid: id }).then(() => {
       const mine_guids = uniq(this.props.partyRelationships.map(({ mine_guid }) => mine_guid));
-      this.props.fetchMineBasicInfoList([...mine_guids]);
+      this.props.fetchMineBasicInfoList([...mine_guids]).then(() => {
```

Comment:
Looks like `mine_guids` is an array? Spreading an array inside of an array cancels each other out.
`(mine_guids)` should be equivalent

Also, it looks like `partyRelationships` and `basicMineInfoListHash` are being passed as optional props AND being fetched on mount. If that's true, we should only fetch if the props were not provided.

Reply:
`mine_guids` was fixed. 

Chatted to Nathan about the props in person.